### PR TITLE
Fix ngen image for building t-route

### PR DESCRIPTION
**Pull request by robertbartel**
Modifying image to get t-route built successfully, so that it is possible to run a NextGen execution that includes t-route routing.

Partially (but not completely) addresses #472; i.e., the image will successfully run serial `ngen` execution with at least some valid configurations of t-route.  Remaining problematic cases _appear_ to be not directly related to the worker image itself.

### CODE REVIEW

File: docker/main/ngen/Dockerfile [hunk 1]
```diff
@@ -374,32 +394,37 @@ RUN echo "export HYDRA_HOST_FILE=${HYDRA_HOST_FILE}" >> /etc/profile \
...
+RUN if [ "${NGEN_WITH_ROUTING}" == "ON" ]; then \
+      # Get subset of requirements needed first (including version specifiers if present) because of netCDF4 handling \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i numpy >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pandas | grep -v geo >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pyyaml >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i cython >> first_reqs.txt ; \
+      cat ${WORKDIR}/t-route-requirements.txt | grep -i pyarrow >> first_reqs.txt ; \
+      pip3 install --no-cache-dir -r first_reqs.txt ; \
+      rm first_reqs.txt ; \
+      # Then install a few more things, including a particular blosc2 version and an adjusted build of tables \
```
Comment (aaraney) [inline docker/main/ngen/Dockerfile:417]: > Looks like [`fiona`](https://pypi.org/project/fiona/1.9.5/) is the culprit. `geopandas` depends on `fiona` and `fiona` by default will install from `pip` with a pre-built `libgdal`, _if it can_. It seems like the issue is `fiona` does not have a distribution with pre-build `libgdal` for the particular target.
> 
> > Fiona has several [extension modules](https://docs.python.org/3/extending/extending.html) which link against libgdal. This complicates installation. Binary distributions (wheels) containing libgdal and its own dependencies are available from the Python Package Index and can be installed using pip.

[source](https://github.com/NOAA-OWP/DMOD/issues/475#issuecomment-1883705630)

```suggestion
      # Then install a few more things, including a particular blosc2 version and an adjusted build of tables \
      dnf install -y gdal-devel ; \
```

**Review by aaraney — CHANGES_REQUESTED** [review]
Only needs a minor change to add support for arm. Looks good otherwise!

Aside, I don't know if we want to address this here or not, but on line 13 of the Dockerfile, `ARG NGEN_BUILD_CONFIG_TYPE="Release"` is not causing `ngen`, `ngen-serial`, or `ngen-parallel` to be built in Release mode. See the `rocky_build_ngen` build stage line ~565.

File: docker/main/ngen/Dockerfile (hunk 1 above)
Comment (robertbartel) [reply docker/main/ngen/Dockerfile:417]: I fixed this differently, by including `gdal-devel` in the main step where the list of packages to be installed via `dnf` is defined.  It'll be installed now in an ancestor build stage of the one the suggestion would have been in.

**Comment by robertbartel** [comment]
I addressed the `gdal-devel` item and fixed things with the value intended for `CMAKE_BUILD_TYPE`.  Should be ready for re-review now.

**Review by aaraney — APPROVED** [review]
Looks good to me! Thanks for making those changes. Merge at will.

**Pull request merged**